### PR TITLE
Check Phase equality directly without accessing map

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This PR removes the unnecessary map access when checking equality of PodPhase.

Thanks to @Miciah who spotted the potential for this change over #84075

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by Miciah** [comment]
It isn't obvious how we should handle the case where a pod is in a phase that is not in the map.  For example, using the following map:
```go
	m := map[v1.PodPhase]int{v1.PodRunning: 0, v1.PodUnknown: 1, v1.PodPending: 2}
```
`PodFailed` and `PodRunning` map to the same value (0), but using the following map:
```go
 	m := map[v1.PodPhase]int{v1.PodPending: 0, v1.PodUnknown: 1, v1.PodRunning: 2}
```
`PodFailed` and `PodPending` map to the same value.

The absence of `PodFailed` and `PodSucceeded` in these maps looks like an oversight.  I don't know whether we need to preserve the current behavior with respect to `PodFailed` and `PodSucceeded` values, but we do need to figure out what the correct behavior of these functions is.

File: pkg/controller/controller_utils.go [hunk 1]
```diff
@@ -832,7 +838,13 @@ func (s ActivePodsWithRanks) Less(i, j int) bool {
 		return len(s.Pods[i].Spec.NodeName) == 0
 	}
 	// 2. PodPending < PodUnknown < PodRunning
-	if podPhaseToOrdinal[s.Pods[i].Status.Phase] != podPhaseToOrdinal[s.Pods[j].Status.Phase] {
+	if _, ok := podPhaseToOrdinal[s.Pods[i].Status.Phase]; !ok {
+		klog.Errorf("phase %v is not in map", s.Pods[i].Status.Phase)
+	}
+	if _, ok := podPhaseToOrdinal[s.Pods[j].Status.Phase]; !ok {
+		klog.Errorf("the phase %v is not in map", s.Pods[i].Status.Phase)
```
Comment (Miciah) [inline pkg/controller/controller_utils.go]: Should be `s.Pods[j].Status.Phase`.  But I'm not sure whether logging an error is the correct behavior.

Comment (tedyu) [reply pkg/controller/controller_utils.go]: I am watching e2e test to see if any such log shows up.

**Comment by tedyu** [comment]
I checked the log for this run:
https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/84198/pull-kubernetes-e2e-gce/

There was no occurrence of the additional logging.

@Miciah 
How about keeping the logs so that we would have better idea if the PodFailed or PodSucceeded ever appears in comparison ?

**Comment by tedyu** [comment]
In the case that either PodFailed or PodSucceeded appears as value for Status.Phase, we can place its precedence behind those of (PodPending < PodUnknown < PodRunning).
Meaning, podPhaseToOrdinal can be defined as:
```
	podPhaseToOrdinal = map[v1.PodPhase]int{v1.PodPending: 0, v1.PodUnknown: 1, v1.PodRunning: 2, v1.PodFailed: 3, v1.PodSucceeded: 4}
```
This way, there is no need to check whether the Status.Phase is in the map or not.

@Miciah 
What do you think ?

**Comment by tedyu** [comment]
@Miciah 
Can you take another look ?

**Pull request closed without merging**